Control.c

```diff
@@ -44,25 +56,59 @@ int main(void)
 	
 	while(1)
 	{
-		// Clear out the screen, and update
-		GLCD_ClearScreen();
-		LCD_UpdateScreenStatus();
-		LCD_UpdateScreenInsulin();
-		
-		// Wait for a short period of time before updating
-		for(i = 0; i < 150000; i++)
+		switch(Control_GlobalState)
 		{
-			for(j = 0; j < 25; j++);
+			case Undefined:
+			case Administration:
+				// Clear out the screen, and update
+			GLCD_ClearScreen();
+			LCD_UpdateScreenStatus();
+			LCD_UpdateScreenState();
+			LCD_UpdateScreenInsulin();
+			
+			// Wait for a short period of time before updating
+			for(i = 0; i < 150000; i++)
+			{
+				for(j = 0; j < 25; j++);
+			}
+				break;
+			case Empty:
+				do {
+					empty = Joystick_GetState();
+				} while(empty != 0x00800000);
+				Control_GlobalStatus = Backward;
+				Control_GlobalState = Administration;
+				break;
+			case Full:
+				do {
+					full = Joystick_GetState();
+				} while(full != 0x02000000);
```

Comment:
Same as above. Another thing to note is that you have case Full: as your empty case and case Empty: as your full case. So they might be backward anyway.